app-backend/backsplash/src/main/scala/services/AnalysisService.scala

```diff
@@ -0,0 +1,108 @@
+package com.azavea.rf.backsplash.analysis
+
+import java.security.InvalidParameterException
+import java.util.UUID
+
+import cats.data.Validated._
+import cats.effect.{IO, Timer}
+import com.azavea.maml.eval.BufferingInterpreter
+import com.azavea.rf.authentication.Authentication
+import com.azavea.rf.backsplash._
+import com.azavea.rf.backsplash.maml.BacksplashMamlAdapter
+import com.azavea.rf.backsplash.parameters.PathParameters._
+import com.azavea.rf.common.RollbarNotifier
+import com.azavea.rf.database.ToolRunDao
+import com.azavea.rf.database.filter.Filterables._
+import com.azavea.rf.database.util.RFTransactor
+import com.azavea.rf.tool.ast.{MapAlgebraAST, _}
+import doobie.implicits._
+import geotrellis.raster._
+import geotrellis.raster.render._
+import geotrellis.server.core.maml._
+import org.http4s.{MediaType, _}
+import org.http4s.dsl._
+import org.http4s.headers._
+
+import scala.util._
+
+class AnalysisService(
+                       interpreter: BufferingInterpreter = BufferingInterpreter.DEFAULT
+                     )(implicit t: Timer[IO])
+  extends Http4sDsl[IO]
+  with RollbarNotifier
+  with Authentication {
+
+  implicit val xa = RFTransactor.xa
+
+  // TODO: DRY OUT
+  object TokenQueryParamMatcher
+    extends QueryParamDecoderMatcher[String]("token")
+
+  object NodeQueryParamMatcher
+    extends OptionalQueryParamDecoderMatcher[String]("node")
+
+  object VoidCacheQueryParamMatcher
+    extends QueryParamDecoderMatcher[Boolean]("voidCache")
+
+  implicit class MapAlgebraAstConversion(val rfmlAst: MapAlgebraAST)
+    extends BacksplashMamlAdapter
+
+  val service: HttpService[IO] =
+    HttpService {
+      case GET -> Root / UUIDWrapper(analysisId) / histogram
+        :? TokenQueryParamMatcher(token)
+        :? NodeQueryParamMatcher(node)
+        :? VoidCacheQueryParamMatcher(void) => {
+
+        ???
+      }
+
+
+      case GET -> Root / UUIDWrapper(analysisId) / IntVar(z) / IntVar(x) / IntVar(y)
+        :? NodeQueryParamMatcher(node) => {
+
+        logger.info(s"Requesting Analysis: ${analysisId}")
+        val tr = ToolRunDao.query.filter(analysisId).select.transact(xa)
+
+        val mapAlgebraAST = tr.flatMap{ toolRun =>
+          logger.info(s"Getting AST")
+          val astOption = toolRun.executionParameters.as[MapAlgebraAST].right.toOption
+          (astOption, node) match {
```

Comment:
Getting a non-exhaustive match here

Reply:
Oh I see the issues you opened here now nvm

Reply:
I opened the issues after my update actually, so it was still good to point these out